File: code401challenges/src/test/java/tree/BinarySearchTreeTest.java
package tree;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

public class BinarySearchTreeTest {
    BinarySearchTree testEmpty;
    BinarySearchTree testFull;
    Node node;

    @Before
    public void setUp () {
        testEmpty = new BinarySearchTree();
        testFull = new BinarySearchTree (node = new Node (9));
    }

    @Test
    public void testTreeConstructor_EmptyTree () {
        Assert.assertEquals(null, testEmpty.root);
    }

    @Test
    public void testTreeConstructor_WithRootNode () {
        Assert.assertEquals(9, testFull.root.value);
    }

    @Test
    public void testAdd_ToEmptyTree () {
        testEmpty.add(1);
        Assert.assertEquals(1, testEmpty.root.value);
    }

    @Test
    public void testAdd_SmallerValue () {
        testFull.add(8);
        Assert.assertEquals(8, testFull.root.left.value);
    }

    @Test
    public void testAdd_LargerValue () {
        testFull.add(10);
        Assert.assertEquals(10, testFull.root.right.value);
    }
}

File: js401challenges/__tests__/BinarySearchTree.test.js
'use strict';

const BinarySearchTree = require('../Trees/BinarySearchTree.js');
const Node = require('../Trees/Node.js');

describe('Binary Search Tree tests', () => {
  let testEmpty;
  let testFull;

  beforeEach(() => {
    testEmpty = new BinarySearchTree();
    testFull = new BinarySearchTree(new Node(9));
  });

  it('can instantiate empty tree', () => {
    expect(testEmpty.root).toBe(null);
  });

  it('can instantiate tree with a root node', () => {
    expect(testFull.root.value).toBe(9);
  });

  it('can add a node to an empty tree', () => {
    testEmpty.add(1);
    expect(testEmpty.root.value).toBe(1);
  });

  it('can add a smaller value than the root', () => {
    testFull.add(8);
    expect(testFull.root.left.value).toBe(8);
  });

  it('can add a larger value than the root', () => {
    testFull.add(10);
    expect(testFull.root.right.value).toBe(9);
  });

  it('contains works when value is at the root', () => {
    expect(testFull.contains(9)).toBe(true);
  });

  it('contains fails when value is not in tree', () => {
    expect(testFull.contains(17)).toBe(false);
  });

});

